index.js

```diff
@@ -1,23 +1,14 @@
 #!/usr/bin/env node
 
-// SUPER IMPORTANT:
-// Node is extremely annoying and refuses to allow ESM imports in
-// CLI tools, so we have to use require here. Do not change to import.
-const fs = require('node:fs').promises;
-const path = require('node:path');
-
-let https;
-try {
-  https = require('node:https');
-} catch (error) {
-  console.error('https support is disabled!');
-}
+import fs from 'node:fs';
+import https from 'node:https';
+import path from 'node:path';
```

Comment:
`ReferenceError: __dirname is not defined`

```suggestion
import path from 'node:path';
import { fileURLToPath } from 'node:url';
    
const __dirname = path.dirname(fileURLToPath(import.meta.url));
```

Reply:
https://github.com/nodejs/node/pull/48740
We can just do import.meta.dirname now

Reply:
Yeah, because that's still kinda new I went with the older version to support more Node versions. Since the whole point of this repo is for changing node versions.